src/wg/image/transform.d

```diff
@@ -49,6 +50,89 @@ Image stripMetadata(Image)(ref Image image) if (isImageBuffer!Image)
 // TODO: flip (not in-place, buffer)
 // TODO: rotation (requires destination image buffer, with matching format)
 
+
+///
+void copy(SrcImg, DestImg)(auto ref SrcImg src, ref DestImg dest) nothrow @nogc if (isImage!SrcImg)// TODO: if (something about dest...)
+{
+    // TODO: assert dest is a writable image
+
+    // strongly type the dest buffer if it's soft-typed
+    static if (is(DestImg == ImageBuffer))
+        auto dst = Image!(ElementType!Src)(dest);
+    else
+        alias dst = dest;
+
+    assert(src.width == dst.width && src.height == dst.height);
+
+    enum srcByRow = __traits(compiles, src.row(0));
+    enum destByRow = __traits(compiles, dst.row(0));
+
+    foreach (y; 0 .. src.height)
+    {
+        static if (srcByRow && destByRow)
+        {
+            dst.row(y)[] = src.row(y)[];
+        }
+        else static if (srcByRow)
+        {
```

Comment:
may be add something like 
```
static assert(__traits(compiles, dst.at(x, y) = srcRow[x];), "DestImg have no `at(x, y)` member providing random access to image data.");
```
?

Reply:
It must have `at` or `row`, or it's not `isImage`.

Reply:
In this scope only `at` needed

Reply:
I think the tests you seek are there... maybe they're just not clear.
`isImage` tests that there is either `at` or `row`, and then above I test for row... in these cases where there is not `row`, I can assume `at`, otherwise the template constraint would fail.

Reply:
All that said, I think `at` and `row` are not very efficient, and I have some plans to replace them with something else.
I mean, maybe keep `at` for low-performance procedural image generation, but these functions need to work in `blocks` of some sort. And I also think the functions should receive the output buffers than to return them, that way the same output buffer can flow down a UFCS chain without doing heaps of mem copies.

Reply:
Probably I wasn't clear but only `SrcImg` is tested by `isImg` and `DstImg` isn't:
```
void copy(SrcImg, DestImg)(auto ref SrcImg src, ref DestImg dest) nothrow @nogc if (isImage!SrcImg)// TODO: if (something about dest...)
```
I just suggested what could be done to implement `TODO: if (something about dest..)` not in signature constraint but in function body directly. But it was suggestion only, ignore it safely.

Reply:
Oh I see, I'm sorry.
Yeah I think it would be better to reject it at the signature... We need to define the kinds of images and what makes them. I think that needs a document with lots of options and conversation.